README.md

```diff
@@ -1,30 +1,145 @@
+
+```sh
+cargo build --profile opt-dev
+```
+
+To build just librav1d using a stable compiler:
+
+```sh
+cargo +stable build --lib --release
+```
+
+## Feature Flags
+
+The following feature flags are supported:
+
+* `asm` - Enables optimized assembly routines, if available for the target
+  platform.
+* `bitdepth_8` - Enables support for 8 bitdepth decoding.
+* `bitdepth_16` - Enables support for 10 and 12 bitdepth decoding.
+
+All of these features are enabled by default. In order to build a version of
+librav1d that disables one or more of these features use the
+`--no-default-features` flag in combination with the `--features` flag to enable
+any desired features. For example, to build without assembly routines, which is
+useful when testing the Rust fallback functions, do the following:
+
+```sh
+cargo build --no-default-features --features="bitdepth_8,bitdepth_16"
+```
+
+## Cross-Compiling
+
+rav1d can be cross-compiled for a target other than the host platform using the
+Cargo `--target` flag. This will require passing additional arguments to the
+Rust compiler to tell it what linker to use. This can be done by setting the
+`RUSTFLAGS` enviroment variable and specifying the `linker` compiler flag. For
+example, compiling for `aarch64-unknown-linux-gnu` from a Ubuntu Linux machine
+would be done as follows:
+
+```sh
+RUSTFLAGS="-C linker=aarch64-linux-gnu-gcc" cargo build --target aarch64-unknown-linux-gnu
+```
+
+If you're cross-compiling in order to run tests under QEMU you'll also need to
+specify the `+crt-static` target feature.
+
+```sh
+RUSTFLAGS="-C target-feature=+crt-static -C linker=aarch64-linux-gnu-gcc" cargo build --target aarch64-unknown-linux-gnu
+```
+
+This will require installing the appropriate cross-platform compiler and linker
+toolchain for your target platform. Examples of how we cross-compile rav1d in CI
+can be found in `.github/workflows/build-and-test-qemu.yml`.
 
 ## Running Tests
 
-Currently we use the original Meson test suite for testing the Rust port. To setup and run these tests, do the following:
+Currently we use the original [Meson](https://mesonbuild.com/) test suite for
+testing the Rust port. This means you'll need to [have Meson
+installed](https://mesonbuild.com/Getting-meson.html) to run tests.
+
+To setup and run the tests, do the following:
 
-First, build the Rust project using Cargo. You'll need to do this step manually before running any tests because it is not built automatically when tests are run. Note that we build with the `--release` flag, adjust paths accordingly
-for debug or cross-target builds.
+First, build the Rust project using Cargo. You'll need to do this step manually
+before running any tests because it is not built automatically when tests are
+run. It's recommended to run tests with either the release or opt-dev profile as
+the debug build runs slowly and often causes tests to timeout. The opt-dev
```

Comment:
```suggestion
First, build `rav1d` using `cargo`. You'll need to do this step manually
before running any tests because it is not built automatically when tests are
run. It's recommended to run tests with either the `release` or `opt-dev` profile as
the debug build runs slowly and often causes tests to timeout. The `opt-dev`
```